compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -104,6 +104,9 @@ pub(crate) fn optimize_into_acir(
         "After `static_assert` and `assert_constant`:",
     )?
     .try_run_pass(Ssa::unroll_loops_iteratively, "After Unrolling:")?
+    // We unroll small Brillig loops based off a comparison between the number of unrolled instructions
+    // and the loop overhead. Loop invariant code motion can affect this calculation so we run it after unrolling.
+    .run_pass(Ssa::loop_invariant_code_motion, "After Loop Invariant Code Motion:")
```

Comment:
Wouldn't `loop_invariant_code_motion` be expected to have a positive effect on unrolling small Brillig loops? I.e. we'd be able to have more small loops unrolled because their loop bodies could get smaller from `loop_invariant_code_motion`?

Reply:
Yeah good catch. I had moved it as with the original pass I wrote `slice_loop` was marking one less loop as small. But as you pointed out this is actually indicative of a deeper error as it would positively effect unrolling small loops. The error looks to be in how I was mapping values, thus why normalizing IDs also fixed the regression. I have moved it back to be before unrolling without a regression.